components/navigation/navChild.js

```diff
@@ -65,24 +65,53 @@ const NavChild = ({ slug, page, color, className }) => {
   }
 
   let link;
+  let divider;
   let icon;
   let target;
-  let url = page.url;
+  let url;
+  if (typeof page.url !== "undefined") {
+    url = page.url;
+  } else {
+    url = "";
+  }
 
-  const isLocalPage = page.url.startsWith("/");
+  const isLocalPage = url.startsWith("/");
+  const isCrossLinkedPage = url.startsWith("https://docs.streamlit.io");
+  const isDivider = url === "";
 
-  if (!isLocalPage) {
+  if (!isLocalPage && !isCrossLinkedPage && !isDivider) {
     icon = <i className={styles.ExternalIcon}>open_in_new</i>;
     target = "_blank";
   }
 
-  if (page.isVersioned && version && isLocalPage) {
+  if (isCrossLinkedPage) {
+    url = url.replace("https://docs.streamlit.io", "");
+    icon = <i className={styles.CrossLinkedIcon}>link</i>;
+  }
+
+  if (page.isVersioned && version && isLocalPage | isCrossLinkedPage) {
     // We need to version this URL, check if the URL has a version for this version
-    const newSlug = page.url.split("/");
+    const newSlug = url.split("/");
     newSlug[0] = version;
     url = `/${newSlug.join("/")}`;
   }
 
+  if (page.name == "---") {
+    divider = (
+      <span className={styles.LinkContainer}>
```

Comment:
Can we use a `<div>` here instead of `<span>`?

Reply:
Update will show in next commit.